containers/src/Kokkos_DynamicView.hpp

```diff
@@ -564,7 +551,23 @@ class DynamicView : public Kokkos::ViewTraits<DataType, P...> {
       m_chunks_host.template allocate_with_destroy<device_space>(
           label, m_chunks.get_ptr());
       m_chunks_host.initialize();
-      m_chunks_host.deep_copy_to(m_chunks);
+
+      // Add some properties if not provided to avoid need for if constexpr
+      using alloc_prop_input = Kokkos::Impl::ViewCtorProp<Prop...>;
+      using alloc_prop       = Kokkos::Impl::ViewCtorProp<
+          Prop..., std::conditional_t<alloc_prop_input::has_execution_space,
+                                      std::integral_constant<unsigned int, 15>,
```

Comment:
Where does this '15' come from?

Reply:
oh yeah I overlooked this at first. this seems a bit hacky lol. I get what this is doing but really? What if we want to check in ViewCtorProp that all the arguments you get are actually expected. Maybe at least make it void_t?

Reply:
It's a random integer that we haven't used before so that we can add a dummy argument. This approach is used pretty much everywhere where you need to make sure that `ViewCtorProp` has a certain property.

Reply:
https://github.com/kokkos/kokkos/blob/5c7a4e3e36912201cbb3a2ec3fe2f3e661c77e84/core/src/impl/Kokkos_ViewCtor.hpp#L115-L126